drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -280,7 +280,7 @@ local function device_init(driver, device)
       deadband_read:merge(clusters.Thermostat.attributes.MinSetpointDeadBand:read())
       device:send(deadband_read)
     end
-    device:set_field(setpoint_limit_device_field.MIN_SETPOINT_DEADBAND_CHECKED, true)
+    device:set_field(setpoint_limit_device_field.MIN_SETPOINT_DEADBAND_CHECKED, true, {persist = true})
```

Comment:
I think we could move this read back to the configure or added function unless there is a good reason to have the deadband read once on init. Typically, does this kind of `once on init` handling is to provide some cleanup for devices that are already in the field and onboarded. However, in this case, devices in the field that are already onboarded would have already had the deadband read during either doConfigure (prior to this change) or in init (recurring on each init with this change). So, I think it makes more sense for this to be in one of the one-time events anyway, such as added or configure

However, given the tight deadline this can be a patch for now that we can move to configure later as needed.

Reply:
Also, it probably makes more sense for this flag to be set once we actually read the value, not just when we send the request.

Reply:
I think I will just move this to do_configure to simplify things. This is where it was in the past so it will be a simple change to move it back. I'm doing a test to confirm that this fixes the issue right now and then I will push up the changes.

Reply:
After discussion, we will leave this read request in `device_init` but move setting the `MIN_SETPOINT_DEADBAND_CHECKED` field to the `MinSetpointDeadBand` attribute handler.